### DISCUSSION TITLE
ブランチ切って、何度かコミットして、プルリクエスト送って、マージされるまでの流れがわからない

### DISCUSSION THREAD

**Discussion by pss-aileen** [first-contributions-ja/first-contributions-ja.github.io#12] [10: 技術についての質問] [answered]
- ブランチを分けて作業をする際、こまめにコミットをしてログを残す...？
- そのコミットメッセージはどのように書けばよいのか←ここがわからない
- 1issue、1commitとするためにはどのような動きをすれば良いのか？
  - `git rebase`とか...？
- 一連の流れの例がほしい。

---
### 以下試行錯誤メモ

```bash
git reset --hard
git reset --soft
git reset --mixed
```
などがあったり
プルリクエストのマージの時に
Squash and merge なるもので一応1コミットにできた
[Test by pss-aileen · Pull Request #2 · pss-aileen/practice-github](https://github.com/pss-aileen/practice-github/pull/2)

**Comment by kazzyfrog** [comment]
1つ簡単なイシュー取り組んで、PR出してみました！
#14  

コミットの粒度などは、初めはそんなに気にしなくても良いのかなとも思いました
なので、細かくコミットして複数になっても気にせずOKです！

**Reply by pss-aileen** [reply]
ありがとうございます！！

私は一人でするときに途中経過すらもコミットしているところがあって、
コミットの粒度がものすごく小さかったんですよね。

この共同プロジェクでは可能であれば、 @kazzyfrog さんがされたように、
1つのコミットにまとめて？ or 作業が終わってからコミット、プルリクを出すといった感じで良いでしょうか？

**Reply by pss-aileen** [reply]
あと、プルリクにあたってのテンプレは kazzyfrogさんのを参考にやってみようと思います

```markdown
## 概要
## 関連するIsssue
```

**Reply by pss-aileen** [reply]
質問攻めで申し訳ないのですが、レビュー機能？

![スクリーンショット 2024-03-12 17 47 55](https://github.com/first-contributions-ja/first-contributions-ja/assets/140990100/1d512922-d912-4f7c-ab91-2526b7a4e25e)

こちらは基本的に、誰か一人がレビューしないとマージできないといった感じでしょうか？

そして、マージは基本「Merge pull request」で良いのでしょうか。
SquashとかRebaseなどもありますが...

<img src="https://github.com/first-contributions-ja/first-contributions-ja/assets/140990100/82b40cf4-bbd9-4c3e-aa4e-9a7dbba1d07c" width="300">

**Comment by kazzyfrog** [comment]
> 1つのコミットにまとめて？ or 作業が終わってからコミット、プルリクを出すといった感じで良いでしょうか？

これは、１つのissueで１つのタスクなら、１コミットにまとめて良いと思います！
１つのイシュー内に、以下のように複数のToDoがあれば、タスクごとにコミットを分けたほうが見やすいかもです：
- [ ] taskA
- [ ] taskB

**Comment by kazzyfrog** [comment]
> こちらは基本的に、誰か一人がレビューしないとマージできないといった感じでしょうか？

これは、リポジトリの設定で、レビューが必要がどうか、最低何人のレビューが必要か、などの設定ができます。
現在は、1人以上のレビュー・承認がないとmainにはマージできない設定になってます！
よかったらレビューしてみて下さい！
今回はテンプレ通りに作成したファイルなので、メッセージは「LGTM」でOKです

あと、マージは基本「Merge pull request」でOKです！

**Reply by pss-aileen** [reply]
ログ...

![スクリーンショット 2024-03-12 18 10 12](https://github.com/first-contributions-ja/first-contributions-ja/assets/140990100/3db14beb-fabe-424d-9d0d-a1b63e786d44)

---

今回はApproveかな...

![スクリーンショット 2024-03-12 18 10 51](https://github.com/first-contributions-ja/first-contributions-ja/assets/140990100/aab781d9-7a36-4d73-9c3e-3f507a197be5)

---

Approveした...！

![スクリーンショット 2024-03-12 18 11 18](https://github.com/first-contributions-ja/first-contributions-ja/assets/140990100/19492d11-7350-48bf-bf54-eaee2a2c447f)

---

![スクリーンショット 2024-03-12 18 11 51](https://github.com/first-contributions-ja/first-contributions-ja/assets/140990100/2c100a37-6e1f-4ff6-a873-c654b4658b3b)

---

![スクリーンショット 2024-03-12 18 12 24](https://github.com/first-contributions-ja/first-contributions-ja/assets/140990100/7c4c017a-c307-4d2d-96b3-f5340df2c5fb)

**Comment by pss-aileen** [comment, marked as answer]
## まとめ

- ブランチ作る
- 1つのissueに1タスクであれば1コミットにまとめる [ref](https://github.com/first-contributions-ja/first-contributions-ja/discussions/12#discussioncomment-8757356)
- プルリクエストを出す [ref](https://github.com/first-contributions-ja/first-contributions-ja/discussions/12#discussioncomment-8757065)
- 誰か一人がレビューする [手順](https://github.com/first-contributions-ja/first-contributions-ja/discussions/12#discussioncomment-8757412)
- Merge pull requestでマージする

**Reply by pss-aileen** [reply]
@kagomen @kana1166
実際のブランチ切ってからプルリク、マージするまでの流れです！！
参考までに共有です！
